src/objc/ModalWebViewController.mm

```diff
@@ -84,27 +84,46 @@ - (void)viewDidDisappear:(BOOL)animated {
   }
 }
 
-- (void)getBrowserCookiesForCurrentUrlWithCompletion:(void (^)(NSDictionary *))completion {
+- (void)getBrowserCookiesForUrlWithCompletion:(NSString *)domain completion:(void (^)(NSDictionary *))completion {
   NSMutableDictionary *cookieDict = [NSMutableDictionary dictionary];
-  NSURL *url = self.webView.URL;
-  if (url == nil) {
-    completion(cookieDict);
-    return;
-  }
-
-  WKHTTPCookieStore *cookieStore =
-      self.webView.configuration.websiteDataStore.httpCookieStore;
+  WKHTTPCookieStore *cookieStore = self.webView.configuration.websiteDataStore.httpCookieStore;
 
   [cookieStore getAllCookies:^(NSArray<NSHTTPCookie *> *cookies) {
     for (NSHTTPCookie *cookie in cookies) {
-      if ([url.host hasSuffix:cookie.domain]) {
+      // Check if the cookie's domain matches the target domain
+      // This handles both exact matches and subdomain matches
+      if ([cookie.domain hasSuffix:domain] || 
+          [cookie.domain isEqualToString:domain]) {
         [cookieDict setObject:cookie.value forKey:cookie.name];
       }
     }
     completion(cookieDict);
   }];
 }
 
+- (void)getBrowserCookiesForCurrentUrlWithCompletion:(void (^)(NSDictionary *))completion {
```

Comment:
This is confusing, there is too much duplication. There should be only three methods:

- `getBrowserCookiesForCurrentUrl` → should get the current webview url and the call `getBrowserCookiesForUrlWithCompletion`
- getBrowserCookiesForUrl → should take the url as param and then call `getBrowserCookiesForUrlWithCompletion`
- `getBrowserCookiesForUrlWithCompletion` → The only one that accesses the cookie storage

Also, as you mention, cookies are based on domain and not on url, so maybe renaming `getBrowserCookiesForUrlWithCompletion` to `getCookiesForDomainWithCompletion` is better